kernel/src/table_changes/schema_compat.rs

```diff
@@ -0,0 +1,302 @@
+use std::collections::{HashMap, HashSet};
+
+use crate::schema::{DataType, Schema, StructField, StructType};
+
+fn is_nullability_compatible(existing_nullable: bool, read_nullable: bool) -> bool {
+    // The case to avoid is when the read_schema is non-nullable and the existing one is nullable.
+    // So we avoid the case where !read_nullable && existing_nullable
+    // Hence we check that !(!read_nullable && existing_nullable)
+    // == read_nullable || !existing_nullable
+    read_nullable || !existing_nullable
+}
+fn is_struct_read_compatible(existing: &StructType, read_type: &StructType) -> bool {
```

Comment:
I was considering using a `DeltaResult` instead of a bool so we can return better errors about *how* a schema differs. Thoughts?

Reply:
I think that makes sense. Something similar to [ValidateColumnMappings](https://github.com/delta-io/delta-kernel-rs/pull/543/files#diff-f5f51b386115172d98ff93569740709026077cd30a402e977d6884f5fa29042cR86) in https://github.com/delta-io/delta-kernel-rs/pull/543, which returns an `Err` with the offending column name path?